File: code/fcn_invAuGFT.m
function [x,invAuGFT] = fcn_invAuGFT(X,U,Q,alpha) 

if nargin < 4
    alpha = 1-1/sqrt(2);
end

% Inverse AuGFT
I = eye(size(U));
beta = sqrt(alpha*(2-alpha));
invAuGFT = [(I-alpha*(Q*Q.'))*U beta*Q];
x = invAuGFT*X;
end

File: code/appendix/py_augft/fcn_invAuGFT.py
import numpy as np

def fcn_invAuGFT(X, U, Q, alpha=(1-1/np.sqrt(2))):
    
    # Inverse AuGFT
    I = np.eye(U.shape[0])
    beta = np.sqrt(alpha * (2 - alpha))
    invAuGFT = np.hstack(((I - alpha * (Q@Q.T)) @ U, beta*Q))
    x = invAuGFT@X

    return (x, invAuGFT)